agent/charms/testflinger-agent-charm/src/charm.py

```diff
@@ -218,14 +224,15 @@ def _render_systemd_unit(self):
         # Reload systemd units
         systemd.daemon_reload()
 
-    def _on_config_changed(self, _):
+    def on_config_changed(self, _):
         self.unit.status = MaintenanceStatus("Handling config_changed hook")
-        self._check_update_repos_needed()
-        self._write_config_files()
-        self._signal_restart_agent()
+        self.check_update_repos_needed()
+        self.write_config_files()
+        self.render_systemd_unit()
+        self.signal_restart_agent()
```

Comment:
Just checking that the addition of `render_systemd_unit` here is intentional

Reply:
yes! There was a small change needed to the systemd unit template and we need to make sure it gets installed to work properly.  Most of the time, it would probably be overkill to install it on every config change, but it doesn't take long and has no effect if nothing changed.